src/storage/txn/actions/prewrite.rs

```diff
@@ -23,9 +26,10 @@ pub fn prewrite<S: Snapshot>(
     mutation: Mutation,
     secondary_keys: &Option<Vec<Vec<u8>>>,
     is_pessimistic_lock: bool,
-) -> Result<TimeStamp> {
+    need_old_value: bool,
+) -> Result<(TimeStamp, Option<Option<OldValue>>)> {
```

Comment:
`Option<Option<OldValue>>` looks stange, the outer `Option` stands for "whether we need to find the old value" and the inner one stands for "whether we do found an old value".
Wonder how can we improve this further.

Reply:
How about defining OldValue as an enum like
```rust
enum OldValue {
    Value {...},
    NotExist,
}
```